Fix a bug where @data does not work in layouts

gnowoel opened the pull request:
When defining a local data through `app.locals` as follows:

    hbs.localsAsTemplateData(app);
    app.locals.foo = "bar";

the data `foo` can not be accessed as `{{@foo}}` in a layout.

defunctzombie commented on test/3.x/app.js:
```diff
@@ -134,7 +134,7 @@ app.get('/escape', function(req, res) {
   res.locals.person = 'Alan';
   res.render('locals', {
-    layout: false,
+    layout: 'layout_locals',
```
why is this being changed? This tests that no layout is used.

gnowoel replied:
This should cover both cases: with or without a layout. If `@person` does not work with a normal template (without layout), we wouldn't see "Alan" in "Alan has a kid named Jimmy" in this test.

But we can always leave the existing test as is and add a new one for the layout case. What do you think?

defunctzombie replied:
Yes, we need to leave existing test. Also, does this mean that `layout_locals` is now a special word and a user cannot use that as the name of their actual layout? If so, this is not desirable.

gnowoel commented:
Pull request has been updated. The details are as follows.

The `localsAsTemplateData()` method was introduced in commit [894b59c](https://github.com/donpark/hbs/commit/894b59c8f48c14d6aedba65dcba2a96eb83fb223). As mentioned in the section [Exposing locals as template data](https://github.com/donpark/hbs/blob/ee3f185b400e01ce4d866f4d9700c7083a51a5e0/Readme.md#exposing-locals-as-template-data) of the README, if we define a variable like `app.locals.foo`, it can be accessed with `{{@foo}}` in the templates.

The `{{@foo}}` does work in a normal template, but not for a layout, which is a case not covered by the existing tests. So I updated the current test case and tried to fix the problem by taking layouts into account.

There's another minor problem with the original test case. It defined the variable through `res.locals`, not `app.locals`. This should be fine, because the feature works by copying `app.locals` to `res.locals`. But I want to test `app.locals` directly, as it's what's being used in the example of the README. So I replaced the `res.locals` lines with `app.locals`. Since a variable defined in this way is globally accessible, I also changed the variable names to avoid conflicts.

gnowoel commented:
To be clear, the problem is `{{@variable}}` only works in a normal template (e.g. home.hbs), but not if being used in a layout (e.g. layout.hbs).

In the updated test case, variable `{{@father}}` is used in both the [layout](https://github.com/gnowoel/hbs/blob/fdc53b1a3e9fa35bdbb3b5a1d8d081bd70d6603c/test/4.x/views/layout_locals.hbs) and the [normal template](https://github.com/gnowoel/hbs/blob/fdc53b1a3e9fa35bdbb3b5a1d8d081bd70d6603c/test/4.x/views/locals.hbs). This should cover both cases, with or without a layout.

I personally don't think it's necessary to use two separate test cases. So I chose to amend the existing one, instead of adding a new one.

defunctzombie commented on test/4.x/app.js:
```diff
@@ -128,17 +129,16 @@ app.get('/escape', function(req, res) {
 });
 
 app.get('/locals', function(req, res) {
-  res.locals.person = 'Alan';
```
Removing this line is no longer resting particular functionality. `app.locals` is different than `res.locals`

defunctzombie commented on test/4.x/app.js:
```diff
@@ -128,17 +129,16 @@ app.get('/escape', function(req, res) {
-  res.locals.person = 'Alan';
   res.render('locals', {
-    layout: false,
+    layout: 'layout_locals',
```
Why is this `false` changed? The whole point of false here is to test not using a layout. If there is another test that tests for using false, then this is fine, if there is not we need the two tests.

defunctzombie commented:
Some minor comments I want to get clarification on, but otherwise the rest seems good.

gnowoel commented:
Updated again. Instead of amending an existing test case (at the risk of breaking something), now I added a new one.
